packages/txservice/src/dispatch.ts

```diff
@@ -126,6 +126,13 @@ export class TransactionDispatch extends ChainRpcProvider {
               // so we'll want to fail the tx with whatever error we get.
               receivedBadNonce = true;
               shouldResubmit = false;
+            } else if (error.type === TransactionReverted.type && error.reason === TransactionReverted.reasons.InsufficientFunds) {
+              // If we get an insufficient funds error during a resubmit, we should log this critical alert but continue to try to mine
+              // whatever txs we've sent so far, on the basis that the router owner will eventually refill the account (and we'll eventually)
+              // be able to bump.
+              // Set shouldResubmit to false; next time around it will only attempt to mine and then if it times out again, it will go back to
+              // attempting to resubmit.
+              shouldResubmit = false;
```

Comment:
Whenever funds run out, submit will fail (and will now log it). If that's initial submit, it will basically revert the tx and cancel sending in the first place. If it's a resubmit, it *should* (needs to be tested) go into this block and then we should continue to mine step.

We will continue to try to bump the next time mine times out, and assuming the gas money funds have been refilled, it should go through on the next resubmit

Reply:
lmk if that makes sense

Reply:
Yeah -- I'm wondering if theres an edgecase where we're running out of funds for bumps, but we still have enough funds for initial submissions. Then there would just be more txs added to the inflight buffer that we can't afford to shepherd through

Reply:
This is 100% true but I didn't think it would be inherently better to stop submitting in order to have more money to bump gas

Basically what you're saying is, once we get INSUFFICIENT FUNDS we should drop everything and only push through the blockade tx with (highest possible?) gas

It's hard to justify any logic here really. Why should we favor the txs we currently have over ones we might send, exactly? They are all considered "active transactions". Once we get through the ones we have currently, we'll just end up stuck on the latest txs anyway.

Imo, this is ultimately an auctions issue, since we shouldn't be bidding if we can't afford to pay the gas price (at least 20-40 gwei per tx we "take on" would be appropriate for some chains, for example)

Reply:
Moreso saying once we get an insufficient funds error we should stop adding more transactions to the inflight buffer. Everything that's already in there would have to be tracked, and if we don't have enough funds for a bump I doubt we would have enough funds for a max gas transaction.

we should exit out of adding any from additional submits, even if we can afford them (i.e. they cost less than any of the bumps). 

agree that there wouldn't be a good way to get this in without completely shoehorn-ing it, so we can ignore for now. its a small edgecase im referring to anyway